moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1037,28 +1030,27 @@ void RobotState::getRobotMarkers(visualization_msgs::MarkerArray& arr, const std
                                  bool include_attached) const
 {
   ros::Time tm = ros::Time::now();
-  for (std::size_t i = 0; i < link_names.size(); ++i)
+  for (const auto& link_name : link_names)
   {
-    ROS_DEBUG_NAMED("robot_state", "Trying to get marker for link '%s'", link_names[i].c_str());
-    const LinkModel* lm = robot_model_->getLinkModel(link_names[i]);
+    ROS_DEBUG_NAMED("robot_state", "Trying to get marker for link '%s'", link_name.c_str());
+    const LinkModel* lm = robot_model_->getLinkModel(link_name);
     if (!lm)
       continue;
     if (include_attached)
-      for (std::map<std::string, AttachedBody*>::const_iterator it = attached_body_map_.begin();
-           it != attached_body_map_.end(); ++it)
-        if (it->second->getAttachedLink() == lm)
+      for (const auto& attached : attached_body_map_)
```

Comment:
Personally, I would still prefer to keep the type in range-based for loops, especially for maps.
In this case that would be
`for(const std::pair<const std::string, AttachedBody*>& attached : attached_body_map_)`
... or we stay with the original.
Otherwise, the (non-trivial) type of `attached.second` is given in the *header file* only (in the definition of `attached_body_map_`).
Unless you use an IDE that provides this information to you automatically, this is horrible to read.

But if multiple other maintainers agree we should prefer `auto` in loops, I will not insist.

Reply:
good point about ``attached.second`` being very hard to read, i'm ok with reverting this auto change

Reply:
That's true, I agree that `auto` in for loops is the most annoying use of `auto`.  

Something to keep in mind is that clang-tidy doesn't gives you the option of automatically converting to for range loops without using `auto`. The good news is that it doesn't complain if you replace `auto` in an existing for loop, so manually adding the type like you suggested wouldn't be changed again. 

I'll revert that commit, and in the future make a seperate PR that runs `modernize-loop-convert` and then follows up manually and fills in the actual types instead of auto.